src/index.ts

```diff
@@ -1,8 +1,53 @@
-import type { TransformedSource, Transformer, TransformOptions } from '@jest/transform';
+import type { CacheKeyOptions, TransformedSource, Transformer, TransformOptions } from '@jest/transform';
 import type { Config } from '@jest/types';
+import createCacheKey from '@jest/create-cache-key-function';
 import { TsJestTransformer } from 'ts-jest/dist/ts-jest-transformer';
 
+import { NgJestConfig } from './config/ng-jest-config';
+import { stringify } from './utils/json';
+import { JsonableValue } from './utils/jsonable-value';
+
+interface ConfigSetItem {
+  configSet: NgJestConfig;
```

Comment:
Does a `ConfigSetItem` contain a `configSet`? Isn't it more like a `ConfigSetCollection` or some other form of container/wrapper?

Reply:
It is a like a container/wrapper. The purpose of this is whenever our jest transformer discovers a processing jest config which is already inside `configSetITems`, we simply return the cached instance `NgJestConfig` without initializing a new instance.

Reply:
Ok, so what about `ConfigSetCache` or `ConfigSetStore`?

Reply:
LGTM, I will change that

Reply:
Hmm, basically I do not like `CachedConfigSet.configSet` so much. It's like `Car.car`. The configSet itself should already be the reference on the set.

Reply:
Ok let’s use the name `ConfigSetStore`

Reply:
No, actually it's good for me this way.

If you don't mind we just rename the inner `configSet: NgJestConfig` to `ngJestConfig: NgJestConfig`?

Reply:
hmm, maybe like this
```
interface ConfigSetItem {
    ngJestConfig: NgJestConfig;
    jestConfig: JsonObject<Config.ProjectConfig>;
    transformerConfig: JsonObject<Config.ProjectConfig & ParsedConfiguration>; // this is for jest cache key
}
```

Reply:
IMO this is better